### PR TITLE
refactor: continue refactor hooks: useBodyScrollLock, useVirtualElementRef, usePortalContainer

### PR DESCRIPTION
**Pull request by NikitaCG**

### CODE REVIEW

File: src/hooks/index.ts [hunk 1]
```diff
@@ -1 +1,4 @@
 export * from './useOutsideClick';
+export * from './useBodyScrollLock';
+export * from './useVirtualElementRef';
+export * from './usePortalContainer';
```
Comment (amje) [inline src/hooks/index.ts:4]: Let's sort this out

File: src/hooks/useBodyScrollLock/useBodyScrollLock.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 
-export interface BodyScrollLockProps {
+export interface UseBodyScrollLockProps {
```
Comment (amje) [inline src/hooks/useBodyScrollLock/useBodyScrollLock.ts:3]: That's a breaking change. We should keep that alias for compatibility

File: src/hooks/usePortalContainer/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# usePortalContainer
+
+<!--/GITHUB_BLOCK-->
+
+```tsx
+import {usePortalContainer} from '@gravity-ui/uikit';
+```
+
+The `usePortalContainer` hook helps to create portal container
```
Comment (amje) [inline src/hooks/usePortalContainer/README.md:11]: This hook doesn't create a portal container, it returns portal container element in the current scope

File: src/hooks/usePortalContainer/usePortalContainer.ts [hunk 4]
```diff
@@ -2,7 +2,9 @@ import React from 'react';
 
 import {PortalContext} from './PortalProvider';
 
-export function usePortalContainer(): HTMLElement | null {
+export type UsePortalContainerReturnType = HTMLElement | null;
```
Comment (amje) [inline src/hooks/usePortalContainer/usePortalContainer.ts:5]: `UsePortalContainerResult`

File: src/hooks/useVirtualElementRef/useVirtualElementRef.ts [hunk 5]
```diff
@@ -16,17 +16,19 @@ export interface UseVirtualElementRefProps {
     rect?: VirtualElementRect | null;
 
     /**
-     * DOM-context of virual element
+     * DOM-context of virtual element
      */
     contextElement?: Element;
 }
 
+export type UseVirtualElementRefReturnType = React.MutableRefObject<VirtualElement>;
```
Comment (amje) [inline src/hooks/useVirtualElementRef/useVirtualElementRef.ts:24]: `UseVirtualElementRefResult`

File: src/hooks/useVirtualElementRef/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+| Name           | Description                                         |                    Type                    | Default |
+| :------------- | :-------------------------------------------------- | :----------------------------------------: | :-----: |
+| rect           | Position of virtual element in relation to viewport | `{top, bottom, left, right}: {[x]:number}` |         |
+| contextElement | DOM-context of virtual element                      |                 `Element`                  |         |
+
+## Return
+
+| Name | Description              |           Type           |
+| :--- | :----------------------- | :----------------------: |
+| ref  | Virtual element ref link | `React.MutableRefObject` |
```
Comment (amje) [inline src/hooks/useVirtualElementRef/README.md:24]: ref object

File: src/hooks/useVirtualElementRef/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import {useVirtualElementRef} from '@gravity-ui/uikit';
+```
+
+The `useVirtualElementRef` hook helps to create virtual element for popup
+
+## Properties
+
+| Name           | Description                                         |                    Type                    | Default |
+| :------------- | :-------------------------------------------------- | :----------------------------------------: | :-----: |
+| rect           | Position of virtual element in relation to viewport | `{top, bottom, left, right}: {[x]:number}` |         |
```
Comment (amje) [inline src/hooks/useVirtualElementRef/README.md:17]: in relation -> relative to

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.tsx [hunk 8]
```diff
@@ -2,12 +2,12 @@ import React from 'react';
 
 import {CSSTransition} from 'react-transition-group';
 
+import {useBodyScrollLock} from '../../hooks/useBodyScrollLock';
```
Comment (amje) [inline src/components/Modal/Modal.tsx:5]: We can import from index: `../../hooks`

File: src/hooks/useBodyScrollLock/index.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export {useBodyScrollLock} from './useBodyScrollLock';
+export type {BodyScrollLockProps} from './useBodyScrollLock';
```
Comment (amje) [inline src/hooks/useBodyScrollLock/index.ts:2]: New interface should be defined and aliased to old one:
```ts
interface UseBodyScrollLockProps {}

type BodyScrollLockProps = UseBodyScrollLockProps
```

And both should be exported

File: src/hooks/usePortalContainer/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import {usePortalContainer} from '@gravity-ui/uikit';
+```
+
+The `usePortalContainer` hook returns portal container element in the current scope
+
+## Result
+
+| Name      | Description                  |         Type         |
+| :-------- | :--------------------------- | :------------------: |
+| container | Container element for portal | `HTMLElement - null` |
```
Comment (amje) [inline src/hooks/usePortalContainer/README.md:17]: This table is not correct for result. There is no `container` name in result, result itself is the container. Let's describe like this:

```
## Result

Container element for portal. `HTMLElement | null`
```

File: src/hooks/useVirtualElementRef/README.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+| Name           | Description                                      |                    Type                    | Default |
+| :------------- | :----------------------------------------------- | :----------------------------------------: | :-----: |
+| rect           | Position of virtual element relative to viewport | `{top, bottom, left, right}: {[x]:number}` |         |
+| contextElement | DOM-context of virtual element                   |                 `Element`                  |         |
+
+## Return
+
+| Name | Description              |       Type        |
+| :--- | :----------------------- | :---------------: |
+| ref  | Virtual element ref link | `React.RefObject` |
```
Comment (amje) [inline src/hooks/useVirtualElementRef/README.md:24]: Same as for `usePortalContainer`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/hooks/useBodyScrollLock/index.ts (hunk 9 above)
Comment (NikitaCG) [reply src/hooks/useBodyScrollLock/index.ts]: fix

**Review by amje — APPROVED** [review]

**Pull request merged**